engine/time/src/main/java/io/deephaven/time/calendar/Calendars.java

```diff
@@ -152,126 +92,143 @@ private void load(final String businessCalendarConfig)
                 return;
             }
             throw new IllegalArgumentException("Multiple calendars have the same name: name='" + name + "'");
         }
 
-        put(name, cal);
+        map.put(name, cal);
+
+        names = map.keySet().toArray(String[]::new);
+        Arrays.sort(names);
     }
 
     /**
-     * Adds a calendar to the collection from the {@code filePath}
+     * Adds a calendar to the collection from a file.
      *
-     * @param filePath must be xml format
+     * @param file business calendar file
      */
-    public void addCalendarFromFile(final String filePath) {
-        addCalendarFromFile(new File(filePath));
+    public static void addCalendarFromFile(final String file) {
+        addCalendarFromFile(new File(file));
     }
 
     /**
-     * Adds a calendar to the collection from the {@code file}
+     * Adds a calendar to the collection from a file.
      *
-     * @param file must be xml format
+     * @param file business calendar file
      */
-    public void addCalendarFromFile(final File file) {
+    public static void addCalendarFromFile(final File file) {
         if (file.getAbsolutePath().endsWith(".calendar")) {
-            final BusinessCalendar cal = DefaultBusinessCalendar.getInstance(file);
+            final BusinessCalendar cal = BusinessCalendarParser.loadBusinessCalendar(file);
             addCalendar(cal);
         } else {
             throw new UnsupportedOperationException("Calendar file must be in .calendar format");
         }
     }
 
-    @Override
-    public int size() {
-        return calendars.size();
-    }
+    private static File inputStreamToFile(@NotNull InputStream inputStream) throws IOException {
+        File calendarFile = File.createTempFile("temp-file-name", ".calendar");
+        FileOutputStream outputStream =
+                new FileOutputStream(calendarFile);
 
-    @Override
-    public boolean isEmpty() {
-        return calendars.isEmpty();
-    }
+        int read;
+        byte[] bytes = new byte[1024];
 
-    @Override
-    public boolean containsKey(Object key) {
-        return !(key == null || !key.getClass().isAssignableFrom(String.class))
-                && calendars.containsKey(((String) key).toUpperCase());
+        while ((read = inputStream.read(bytes)) != -1) {
+            outputStream.write(bytes, 0, read);
+        }
 
+        outputStream.close();
+        return calendarFile;
     }
 
-    @Override
-    public boolean containsValue(Object value) {
-        return calendars.containsValue(value);
-    }
+    // end region
 
-    @Override
-    public BusinessCalendar get(Object key) {
-        return calendars.get(key);
-    }
+    // region Methods
 
-    @Override
-    public BusinessCalendar put(String key, BusinessCalendar value) {
-        Require.neqNull(key, "key");
-        key = key.toUpperCase();
-        return calendars.put(key, value);
+    /**
+     * Sets the default calendar name.
```

Comment:
What does this mean in the context of a "collection of calendars"?

Reply:
Updated docs